src/importexport/musicxml/internal/musicxml/import/importmusicxmlpass2.cpp

```diff
@@ -8513,6 +8513,8 @@ static void addWavyLine(ChordRest* cr, const Fraction& tick,
                 trill = Factory::createTrill(cr->score()->dummy());
                 trill->setTrack(trk);
                 trill->setTrack2(trk);
+                // To be sure the right voice is set as createTrill does not copy the right voice from cr->score
+                trill->setVoice(cr->voice());
```

Comment:
This comment is not accurate. `createTrill` does nothing with the voice at all, and that is by design. 

`setVoice` actually just calls `setTrack`. Given the definition of `trk`, it looks like it was intentional that the trill is assigned to the first voice (0). If that is not desirable, just replace `trk` with `track` (see above) and remove `trk` if it becomes unused because of that.

Reply:
Ok. It works that way
I don't know why I think trills has a voice property which I think is linked to the chord it is attached to. I think the voice is used to know where it ends and it depends on the voice's chords.

Reply:
Looking at @lvinken's sample score shows that this trill is attached to voice 2: it is green and below staff
![image](https://github.com/user-attachments/assets/c6c93119-3a0f-4d4e-87d1-14a057d62e90)

Before your PR the entire trill gets lost on export already. This happens with master and 4.4.4 (forced to open that 4.5 score) at least, also with 3.7, but not with 3.6.2 (on the extracted mscx with the LastEID line deleted). So this part is a Mu3 regression. 

With your PR a) the trill exports that voice assignment is lost (the trill is blue and shows above staff rather than below).
Same as without your PR

With that
```cpp
                trill->setVoice(cr->voice());
```
the voice assignment gets kept on import (it is green), but it still shows above score, so there's still something amiss

Reply:
However: trying to access the trill's properties leads to a crash!

MuseScore4.exe!mu::inspector::AbstractInspectorModel::styleIdByPropertyId(const mu::engraving::Pid pid) Line 419
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\inspector\models\abstractinspectormodel.cpp(419)
MuseScore4.exe!mu::inspector::AbstractInspectorModel::loadPropertyItem(mu::inspector::PropertyItem * propertyItem, const QList<mu::engraving::EngravingItem *> & elements, std::function<QVariant __cdecl(QVariant const &)> convertElementPropertyValueFunc) Line 640
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\inspector\models\abstractinspectormodel.cpp(640)
MuseScore4.exe!mu::inspector::AbstractInspectorModel::loadPropertyItem(mu::inspector::PropertyItem * propertyItem, std::function<QVariant __cdecl(QVariant const &)> convertElementPropertyValueFunc) Line 623
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\inspector\models\abstractinspectormodel.cpp(623)
MuseScore4.exe!mu::inspector::OrnamentSettingsModel::loadProperties() Line 124
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\inspector\models\notation\ornaments\ornamentsettingsmodel.cpp(124)
MuseScore4.exe!mu::inspector::AbstractInspectorModel::updateProperties() Line 403
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\inspector\models\abstractinspectormodel.cpp(403)
MuseScore4.exe!mu::inspector::AbstractInspectorModel::qt_static_metacall(QObject * _o, QMetaObject::Call _c, int _id, void * * _a) Line 328
	at C:\Users\Jojo\Documents\GitHub\MuseScore\msvc.build\x64-debug\src\inspector\inspector_autogen\include_Debug\IVK4DJNV27\moc_abstractinspectormodel.cpp(328)
[External Code]
MuseScore4.exe!mu::inspector::ElementRepositoryService::elementsUpdated(const QList<mu::engraving::EngravingItem *> & _t1) Line 140
	at C:\Users\Jojo\Documents\GitHub\MuseScore\msvc.build\x64-debug\src\inspector\inspector_autogen\include_Debug\KA2SHC4NR2\moc_elementrepositoryservice.cpp(140)
MuseScore4.exe!mu::inspector::ElementRepositoryService::updateElementList(const QList<mu::engraving::EngravingItem *> & newRawElementList, mu::engraving::SelState selectionState) Line 77
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\inspector\internal\services\elementrepositoryservice.cpp(77)
MuseScore4.exe!mu::inspector::InspectorListModel::setElementList(const QList<mu::engraving::EngravingItem *> & selectedElementList, mu::engraving::SelState selectionState) Line 110
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\inspector\models\inspectorlistmodel.cpp(110)
MuseScore4.exe!mu::inspector::InspectorListModel::updateElementList() Line 340
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\inspector\models\inspectorlistmodel.cpp(340)
MuseScore4.exe!mu::inspector::InspectorListModel::setInspectorVisible(bool visible) Line 152
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\inspector\models\inspectorlistmodel.cpp(152)
MuseScore4.exe!mu::inspector::InspectorListModel::qt_static_metacall(QObject * _o, QMetaObject::Call _c, int _id, void * * _a) Line 70
	at C:\Users\Jojo\Documents\GitHub\MuseScore\msvc.build\x64-debug\src\inspector\inspector_autogen\include_Debug\IVK4DJNV27\moc_inspectorlistmodel.cpp(70)
MuseScore4.exe!mu::inspector::InspectorListModel::qt_metacall(QMetaObject::Call _c, int _id, void * * _a) Line 114
	at C:\Users\Jojo\Documents\GitHub\MuseScore\msvc.build\x64-debug\src\inspector\inspector_autogen\include_Debug\IVK4DJNV27\moc_inspectorlistmodel.cpp(114)
[External Code]
MuseScore4.exe!muse::dock::DockFrameModel::currentDockChanged() Line 367
	at C:\Users\Jojo\Documents\GitHub\MuseScore\msvc.build\x64-debug\src\framework\dockwindow\muse_dockwindow_autogen\include_Debug\EGDWYGFDBT\moc_dockframemodel.cpp(367)
MuseScore4.exe!muse::dock::DockFrameModel::listenChangesInFrame::__l2::<lambda_2>::operator()() Line 158
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\framework\dockwindow\internal\dockframemodel.cpp(158)
MuseScore4.exe!QtPrivate::FunctorCall<QtPrivate::IndexesList<>,QtPrivate::List<>,void,`muse::dock::DockFrameModel::listenChangesInFrame'::`2'::<lambda_2>>::call(muse::dock::DockFrameModel::listenChangesInFrame::__l2::<lambda_2> & f, void * * arg) Line 146
	at C:\Qt\6.2.9\include\QtCore\qobjectdefs_impl.h(146)
MuseScore4.exe!QtPrivate::Functor<`muse::dock::DockFrameModel::listenChangesInFrame'::`2'::<lambda_2>,0>::call<QtPrivate::List<>,void>(muse::dock::DockFrameModel::listenChangesInFrame::__l2::<lambda_2> & f, void * __formal, void * * arg) Line 256
	at C:\Qt\6.2.9\include\QtCore\qobjectdefs_impl.h(256)
MuseScore4.exe!QtPrivate::QFunctorSlotObject<`muse::dock::DockFrameModel::listenChangesInFrame'::`2'::<lambda_2>,0,QtPrivate::List<>,void>::impl(int which, QtPrivate::QSlotObjectBase * this_, QObject * r, void * * a, bool * ret) Line 420
	at C:\Qt\6.2.9\include\QtCore\qobjectdefs_impl.h(420)
[External Code]
MuseScore4.exe!KDDockWidgets::Frame::currentDockWidgetChanged(KDDockWidgets::DockWidgetBase * _t1) Line 296
	at C:\Users\Jojo\Documents\GitHub\MuseScore\msvc.build\x64-debug\src\framework\dockwindow\thirdparty\KDDockWidgets\src\kddockwidgets_autogen\include_Debug\PGVKHOT7RA\moc_Frame_p.cpp(296)
MuseScore4.exe!KDDockWidgets::Frame::qt_static_metacall(QObject * _o, QMetaObject::Call _c, int _id, void * * _a) Line 126
	at C:\Users\Jojo\Documents\GitHub\MuseScore\msvc.build\x64-debug\src\framework\dockwindow\thirdparty\KDDockWidgets\src\kddockwidgets_autogen\include_Debug\PGVKHOT7RA\moc_Frame_p.cpp(126)
[External Code]
MuseScore4.exe!KDDockWidgets::TabWidgetQuick::currentDockWidgetChanged(KDDockWidgets::DockWidgetBase * _t1) Line 229
	at C:\Users\Jojo\Documents\GitHub\MuseScore\msvc.build\x64-debug\src\framework\dockwindow\thirdparty\KDDockWidgets\src\kddockwidgets_autogen\include_Debug\Q7DMEJDZ4X\moc_TabWidgetQuick_p.cpp(229)
MuseScore4.exe!KDDockWidgets::TabWidgetQuick::setCurrentDockWidget(int index) Line 73
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\framework\dockwindow\thirdparty\KDDockWidgets\src\private\quick\TabWidgetQuick.cpp(73)
MuseScore4.exe!KDDockWidgets::TabWidgetQuick::qt_static_metacall(QObject * _o, QMetaObject::Call _c, int _id, void * * _a) Line 100
	at C:\Users\Jojo\Documents\GitHub\MuseScore\msvc.build\x64-debug\src\framework\dockwindow\thirdparty\KDDockWidgets\src\kddockwidgets_autogen\include_Debug\Q7DMEJDZ4X\moc_TabWidgetQuick_p.cpp(100)
MuseScore4.exe!KDDockWidgets::TabWidgetQuick::qt_metacall(QMetaObject::Call _c, int _id, void * * _a) Line 200
	at C:\Users\Jojo\Documents\GitHub\MuseScore\msvc.build\x64-debug\src\framework\dockwindow\thirdparty\KDDockWidgets\src\kddockwidgets_autogen\include_Debug\Q7DMEJDZ4X\moc_TabWidgetQuick_p.cpp(200)
[External Code]
MuseScore4.exe!muse::dock::DockTabBar::event(QEvent * event) Line 49
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\framework\dockwindow\internal\docktabbar.cpp(49)
[External Code]
MuseScore4.exe!KDDockWidgets::MouseEventRedirector::eventFilter(QObject * source, QEvent * ev) Line 90
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\framework\dockwindow\thirdparty\KDDockWidgets\src\private\quick\QWidgetAdapter_quick.cpp(90)
[External Code]
MuseScore4.exe!main(int argc, char * * argv) Line 206
	at C:\Users\Jojo\Documents\GitHub\MuseScore\src\app\main.cpp(206)
[External Code]

Reply:
I think the problem is no related to the patch:

- If you compile a version **without** the patch (OS: Windows 11 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.5.0-, revision: github-musescore-musescore) and then:
1. Create a file in MuseScore with a trill as simple as: [Sample score.zip](https://github.com/user-attachments/files/18138260/Sample.score.zip)
2. Export the file to a .musicxml file
3. Import this .musicxml
4. Try to access the Trill properties.... you also get a crash. 

- However with this version **with** the patch:
1. Export the example MuseScore file to musicxml
5. Imports the MuseScore file and saves it as ".mscz"
6. Opens the ".mscz" file: you can access to the Trill properties

So, I think the problems is related with the version we are dealing with, but not with the patch itself. The problem is unrelated to the patch. With or without the patch:
- If you try to access to any Trill property when you open a .musicxml file... you get a crash
- If you save this .musicxml  to .mscz the file and you open the  ".mscz" file you can access to Trill properties.

It seems that there is a problem editing the Trill properties when you open a .musicxml file. Please, could you double check it @Jojo-Schmitz ?

Regarding 

> added, the voice assignment gets kept on import (it is green), but it still shows above score, so there's still something amiss

It is shown above the score unless you save it as ".mscz" and then opens the ".mscz". In that case you can see the trill below the score.

Reply:
Based on the information you have provided I have added the following change to the patch ( I insist that the problem was already there before the patch) and it seems now to work properly.

```
pacebes@cozuelos:/C/D/MuseScore$ git diff
diff --git a/src/inspector/models/abstractinspectormodel.cpp b/src/inspector/models/abstractinspectormodel.cpp
index 8aeec7087d..3e056c2343 100644
--- a/src/inspector/models/abstractinspectormodel.cpp
+++ b/src/inspector/models/abstractinspectormodel.cpp
@@ -416,10 +416,12 @@ mu::engraving::Sid AbstractInspectorModel::styleIdByPropertyId(const mu::engravi
     mu::engraving::Sid result = mu::engraving::Sid::NOSTYLE;

     for (const mu::engraving::EngravingItem* element : m_elementList) {
-        result = element->getPropertyStyle(pid);
+        if (element) {
+            result = element->getPropertyStyle(pid);

-        if (result != mu::engraving::Sid::NOSTYLE) {
-            break;
+            if (result != mu::engraving::Sid::NOSTYLE) {
+                break;
+            }
         }
     }

```

Could you check that you don't have any problem now ?. 

Two additional questions:
- How did you get the crash information ( _MuseScore4.exe!mu::inspector::AbstractInspectorModel::styleIdByPropertyId(const mu::engraving::Pid pid) Line 419 ..._)
- Is there something pending to be fixed ?

Reply:
> I think the problem is no related to the patch:

Yes the crash happens also with 4.4.4 and others, but only with the MusicXML export of this PR, at least for me.

> It is shown above the score unless you save it as ".mscz" and then opens the ".mscz". In that case you can see the trill below the score.

But that should not happen!

> Could you check that you don't have any problem now ?.

It still crashes, in a Debug build, here:
```cpp
    for (const mu::engraving::EngravingItem* element : elements) {
        IF_ASSERT_FAILED(element) {
            continue;
        }
```
Won't crash in non-Debug builds of course though

Reply:
I, like you, get a crash opening the .musicxml file with the latests 4.4.4 version (_OS: Windows 11 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.4-243461245, revision: github-musescore-musescore-2232670_) when I try to access the properties

But I also get the same problem with 4.4.4 if:
1. I create with 4.4.4 a really simple mscz file with a _trill line_ in the only voice it has: [SimpleScoreWithTrillsOnly1Voice.zip](https://github.com/user-attachments/files/18140059/SimpleScoreWithTrillsOnly1Voice.zip)
2. I export with 4.4.4 the .mscz to .musicxml 
3. I open with 4.4.4 the previously exported (step 2) .musicxml file
4. I try to access with 4.4.4 to _Trill line_ properties

I get a crash. 

So, my points are:
- The crashes already happens with 4.4.4 itself exporting an score with _trills lines_ to musicxml, opening the .musicxml file and trying to access _Trill Lines_ properties. The problem has nothing to do with this patch or the 4.5 version
- We have inherited this problem in 4.5 version and it is unrelated to this patch

As you say the assert failed with the latest patch (no crash in non-debug version). I have been debugging the problems and it fails with the following Pids:
- ARTICULATION_ANCHOR
- INTERVAL_ABOVE
- INTERVAL_BELOW
- ORNAMENT_SHOW_ACCIDENTAL
- ORNAMENT_SHOW_CUE_NOTE
- START_ON_UPPER_NOTE

I don't know if this elements should be present when exporting the trill to .musicxml or if it something else. If would appreciate if you have any hint

Reply:
I have been looking for some kind of "official" musicXML files with "Trill Lines" and I have found the following page https://lilypond.org/doc/v2.25/input/regression/musicxml/collated-files.html with some examples with trill lines: [trill-wavy-lines-lily-examples.zip](https://github.com/user-attachments/files/18140235/trill-wavy-lines-lily-examples.zip)

Two of them do no generate problems and one of them does generate problems.

Reply:
OK, I see, the crash seems indeed unrelated to your PR. I just detected it with it (and then withou it too).

It seems something is wrong wit the import code, I suspect the `Factory::createTrill(cr->score()->dummy());` to not fully create and initialize the trill with all it's properties.

As far as I can tell it only initiates PLACEMENT and OFFSET in
```cpp
//---------------------------------------------------------
//   trillStyle
//---------------------------------------------------------

static const ElementStyle trillStyle {
    { Sid::trillPlacement, Pid::PLACEMENT },
    { Sid::trillPosAbove,  Pid::OFFSET },
};
```

Reply:
Sorry. I deleted the previous comment by mistake.

I have got some kind of _Trill Lines_ examples from: https://lilypond.org/doc/v2.25/input/regression/musicxml/collated-files.html: [trill-wavy-lines-lily-examples.zip](https://github.com/user-attachments/files/18140279/trill-wavy-lines-lily-examples.zip)

MuseScore has problems with all of them. I will try to solve the problem with your hints.

Reply:
Pandora's Box?

Reply:
I think I have found the problem and I have uploaded the latest changes. I have copied the "Ornaments" values from the linked Chord to the Trill. 
I hope it works now as it should.

Reply:
Works fine, except for the glicht that after the import the trill shows above staff, but below after saving as mscz and reopening.
After the import its placement property shows above (which is where it is), after save as mscz and reopen its placement property shows at Auto, and the trill shows below, which is what it should.
When after the import using the "Reset to default", that corrects this too

Reply:
I have made a new change to set this property to Auto.

Reply:
Yessss, that's it!